egs/wsj/s5/steps/nnet3/tdnn/train_raw_nnet.sh

```diff
@@ -0,0 +1,626 @@
+
+! [ "$num_hidden_layers" -gt 0 ] && echo \
+ "$0: Expected num_hidden_layers to be defined" && exit 1;
+
+if $dense_targets; then
+  tmp_num_targets=`feat-to-dim scp:$targets_scp - 2>/dev/null` || exit 1
+
+  if [ $tmp_num_targets -ne $num_targets ]; then
+    echo "Mismatch between num-targets provided to script vs configs"
+    exit 1
+  fi
+fi
+
+if [ $stage -le -4 ] && [ -z "$egs_dir" ]; then
+  extra_opts=()
+  [ ! -z "$cmvn_opts" ] && extra_opts+=(--cmvn-opts "$cmvn_opts")
+  [ ! -z "$feat_type" ] && extra_opts+=(--feat-type $feat_type)
+  [ ! -z "$online_ivector_dir" ] && extra_opts+=(--online-ivector-dir $online_ivector_dir)
+  extra_opts+=(--transform-dir "$transform_dir")
+  extra_opts+=(--left-context $left_context)
+  extra_opts+=(--right-context $right_context)
+  echo "$0: calling get_egs.sh"
+
+  if $dense_targets; then
+    target_type=dense
+  else
+    target_type=sparse
+  fi
+
+  steps/nnet3/get_egs_targets.sh $egs_opts "${extra_opts[@]}" \
+    --samples-per-iter $samples_per_iter --stage $get_egs_stage \
+    --cmd "$cmd" --nj $nj \
+    --frames-per-eg $frames_per_eg \
+    --target-type $target_type --num-targets $num_targets \
+    $data $targets_scp $dir/egs || exit 1;
+fi
+
+[ -z $egs_dir ] && egs_dir=$dir/egs
+
+if [ "$feat_dim" != "$(cat $egs_dir/info/feat_dim)" ]; then
+  echo "$0: feature dimension mismatch with egs, $feat_dim vs $(cat $egs_dir/info/feat_dim)";
+  exit 1;
+fi
+if [ "$ivector_dim" != "$(cat $egs_dir/info/ivector_dim)" ]; then
+  echo "$0: ivector dimension mismatch with egs, $ivector_dim vs $(cat $egs_dir/info/ivector_dim)";
+  exit 1;
+fi
+
+# copy any of the following that exist, to $dir.
+cp $egs_dir/{cmvn_opts,splice_opts,final.mat} $dir 2>/dev/null
+
+# confirm that the egs_dir has the necessary context (especially important if
+# the --egs-dir option was used on the command line).
+egs_left_context=$(cat $egs_dir/info/left_context) || exit -1
+egs_right_context=$(cat $egs_dir/info/right_context) || exit -1
+ ( [ $egs_left_context -lt $left_context ] || \
+   [ $egs_right_context -lt $right_context ] ) && \
+   echo "$0: egs in $egs_dir have too little context" && exit -1;
+
+frames_per_eg=$(cat $egs_dir/info/frames_per_eg) || { echo "error: no such file $egs_dir/info/frames_per_eg"; exit 1; }
+num_archives=$(cat $egs_dir/info/num_archives) || { echo "error: no such file $egs_dir/info/frames_per_eg"; exit 1; }
+
+# num_archives_expanded considers each separate label-position from
+# 0..frames_per_eg-1 to be a separate archive.
+if [ "$chunk_training" == "true" ]; then
+  num_archives_expanded=$num_archives
+else
+  num_archives_expanded=$[$num_archives*$frames_per_eg]
+fi
+
+[ $num_jobs_initial -gt $num_jobs_final ] && \
+  echo "$0: --initial-num-jobs cannot exceed --final-num-jobs" && exit 1;
+
+[ $num_jobs_final -gt $num_archives_expanded ] && \
+  echo "$0: --final-num-jobs cannot exceed #archives $num_archives_expanded." && exit 1;
+
+
```

Comment:
you should probably check that add_lda is defined at this point and die if not.  otherwise we'll get confusing errors if something went wrong in generating the configs, or a different script was used.